examples/deserialize-transaction/src/example.ts

```diff
@@ -0,0 +1,403 @@
+/**
+ * EXAMPLE
+ * Deserialize and inspect a transaction with @solana/web3.js.
+ *
+ * Before running any of the examples in this monorepo, make sure to set up a test validator by
+ * running `pnpm test:live-with-test-validator:setup` in the root directory.
+ *
+ * To run this example, execute `pnpm start` in this directory.
+ */
+import createLogger from '@solana/example-utils/createLogger.js';
+import {
+    Address,
+    appendTransactionMessageInstructions,
+    assertAccountDecoded,
+    assertAccountExists,
+    assertIsInstructionWithAccounts,
+    assertIsInstructionWithData,
+    compressTransactionMessageUsingAddressLookupTables,
+    createKeyPairSignerFromBytes,
+    createSolanaRpc,
+    createTransactionMessage,
+    decodeTransactionMessage,
+    fetchJsonParsedAccount,
+    getAddressEncoder,
+    getBase64EncodedWireTransaction,
+    getBase64Encoder,
+    getCompiledTransactionMessageDecoder,
+    getTransactionDecoder,
+    partiallySignTransactionMessageWithSigners,
+    pipe,
+    setTransactionMessageFeePayer,
+    setTransactionMessageLifetimeUsingBlockhash,
+    verifySignature,
+} from '@solana/web3.js';
+import { getAddMemoInstruction, MEMO_PROGRAM_ADDRESS, parseAddMemoInstruction } from '@solana-program/memo';
+import {
+    getTransferSolInstruction,
+    identifySystemInstruction,
+    parseTransferSolInstruction,
+    SystemInstruction,
+} from '@solana-program/system';
+
+const log = createLogger('Deserialize');
+
+/**
+ * SETUP: LOOKUP TABLE ADDRESS
+ * Our fixtures include a lookup table, which we're going to use for this transaction
+ * This is the address of the lookup table account from `scripts/fixtures/address-lookup-table-account.json`
+ */
+const LOOKUP_TABLE_ADDRESS = '2JPQuT3dHtPjrdcbUQyrrT4XYRYaWpWfmAJ54SUapg6n' as Address;
+log.info({ address: LOOKUP_TABLE_ADDRESS }, '[setup] Setting lookup table address');
+
+/**
+ * SETUP: RPC CONNECTION
+ * While in this example we won't send the transaction, we will use the remote procedure call (RPC) server to:
+ * - fetch the blockhash for the transaction lifetime
+ * - fetch the address lookup table used in the transaction
+ *
+ * This example uses your local test validator which must be running before you run this script.
+ */
+const rpc = createSolanaRpc('http://127.0.0.1:8899');
+
+/**
+ * SETUP: TRANSACTION LIFETIME
+ * We will fetch the latest blockhash, which we will use as the transaction's lifetime
+ * See `examples/transfer-lamports` for more detail on transaction lifetime
```

Comment:
+1. These examples actually end up doing rather a lot so referring to one can be a bit confusing as the reader would then need to peruse the example to find the section that mentions transaction lifetimes.

It might be worth considering smaller examples such as "set blockhash transaction lifetime" and "set durable nonce transaction lifetime" that can be more easily referred by other examples in the docblocs (the code would still be duplicated so the example file is self-sufficient).

Reply:
Yep agreed, it'd be better to have a smaller example to refer to here

The main reason I didn't want to just copy the whole doc block is because some of the detail isn't necessarily relevant, since this example doesn't actually send/confirm the transaction